src/utils/restClient/index.ts

```diff
@@ -0,0 +1,83 @@
+import fetch, { Response, RequestInit } from 'node-fetch';
+import { v4 as uuidv4 } from 'uuid';
+
+const DEFAULT_TIMEOUT = 10000;
+const DEFAULT_MAX_RETRIES = 2;
+const BASE_URL = 'https://api.mercadopago.com';
+
+interface RestClientConfig {
+  timeout?: number;
+  queryParams?: Record<string, string | number>;
+  retries?: number;
+  idempotencyKey?: string;
+}
+
+class RestClient {
+	private static generateIdempotencyKey(): string {
+		return uuidv4();
+	}
+
+	private static appendQueryParamsToUrl(url: string, queryParams?: Record<string, string | number>): string {
+		if (!queryParams) return url;
+
+		const searchParams = new URLSearchParams();
+
+		for (const key in queryParams) {
+			if (Object.prototype.hasOwnProperty.call(queryParams, key)) {
+				searchParams.append(key, queryParams[key].toString());
+			}
+		}
+
+		return url.includes('?') ? `${url}&${searchParams.toString()}` : `${url}?${searchParams.toString()}`;
+	}
+
+	static async fetch<T>(
+		endpoint: string,
+		config?: RestClientConfig & RequestInit
+	): Promise<T> {
+		const {
+			timeout = DEFAULT_TIMEOUT,
+			retries: maxRetries = DEFAULT_MAX_RETRIES,
+			idempotencyKey,
+			queryParams,
+			method = 'GET',
+			...customConfig
+		} = config || {};
+
+		const url = `${BASE_URL}${endpoint}${queryParams ? RestClient.appendQueryParamsToUrl('', queryParams) : ''}`;
```

Comment:
Passando `''` para `appendQueryParamsToUrl` a verificação `url.includes('?')` (linha 31) sempre vai retornar `false`

```suggestion
		const url = RestClient.appendQueryParamsToUrl(`${BASE_URL}${endpoint}`, queryParams);
```
Não deveria ser assim?

Reply:
sim tem razao, entre os varios reescreve-apaga, isso acabou ido errado, mas já corrigi